bigbluebutton-html5/imports/api/meetings/server/publishers.js

```diff
@@ -132,3 +133,37 @@ function timeRemainingPublish(...args) {
 }
 
 Meteor.publish('meeting-time-remaining', timeRemainingPublish);
+
+function notifications() {
+  const tokenValidation = AuthTokenValidation.findOne({ connectionId: this.connection.id });
+  if (tokenValidation || tokenValidation.validationStatus === ValidationStates.VALIDATED) {
+    notificationEmitter.on('notification', (notification) => {
+      const { meetingId, userId } = tokenValidation;
+      switch (notification.type) {
+        case 'notifyAllInMeeting':
+          if (notification.meetingId === meetingId) this.added('notifications', Math.random() * 500, notification);
+          break;
+        case 'NotifyUserInMeeting':
+          if (notification.meetingId === meetingId && notification.userId === userId) this.added('notifications', Math.random() * 500, notification);
+          break;
+        case 'NotifyRoleInMeeting': {
+          const user = Users.findOne({ userId, meetingId }, { fields: { role: 1, userId: 1 } });
+          if (notification.meetingId === meetingId && notification.role === user.role) this.added('notifications', Math.random() * 500, notification);
```

Comment:
The use of `Math.random()` is triggering some alerts in our static code analysis tool. What is the role of this? Spacing out the notifications?

Reply:
it is used to set a random id for each notification, as the ID is required (but not relevant to the implementation) when using `.added`